### PR TITLE
Test handling of moving symlinks around in git-annex

### LINKED ISSUE #46: add a test and verify that fscacher works ok with git-annex'ed files in direct mode
i.e. whenever a file is a symlink pointing to some .git/annex/objects, and we should be using cached value even if original file changes its name and/or location but still points to the same file under `.git/annex/objects`. 
I guess a test could just be done by manipulating symlinks explicitly, but I think it would be best to ensure that there is no some other side-effect from git-annex doing that.  I guess no need to drag `datalad` for this minor test, and we could just use `git annex` and `git` commands directly to

- create a file with content, `git annex add` and `git commit` it
- cache some function result
- `git mv` the file to a new filename and `git commit`, redo the function call and ensure that we do not recompute
- `git mv` the file to another directory and `git commit`, redo the function call and ensure that we do not recompute

related to #44 where we actually don't want to pass resolved path into underlying function. But we still want to use resolved path for getting the fingerprint. So getting this test done would ensure that we do not break it whenever we finally address #44

### PR DESCRIPTION
**Pull request by jwodder**
Closes #46.

### CODE REVIEW

File: src/fscacher/tests/test_cache.py [hunk 1]
```diff
@@ -342,3 +344,47 @@ def test_cache_control_envvar(
     c = PersistentCache(name="test-cache-control-envvar", envvar="MYCACHE_CONTROL")
     assert clear_spy.called is cleared
     assert c._ignore_cache is ignored
+
+
+@pytest.mark.skipif(shutil.which("git-annex") is None, reason="git annex required")
+def test_follow_moved_symlink(cache, tmp_path):
+    calls = []
```
Comment (yarikoptic) [inline src/fscacher/tests/test_cache.py:351]: some test runs seems stuck in yellow...
I don't think we install `git-annex` in CI, we should, at least on linux and osx where we know that symlinks should generally be supported/available

Comment (jwodder) [reply src/fscacher/tests/test_cache.py:351]: git-annex installed, but GitHub Actions seems to be having some queuing/runner availability problem that is keeping the tests from running.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -42,6 +42,14 @@ jobs:
                 python -m pip install --upgrade pip wheel
                 python -m pip install --upgrade --upgrade-strategy=eager tox
 
+            - name: Install git annex (Ubuntu)
+              if: startsWith(matrix.os, 'ubuntu')
+              run: sudo apt-get update && sudo apt-get install -y git-annex
+
+            - name: Install git annex (macOS)
+              if: startsWith(matrix.os, 'macos')
+              run: brew install git-annex
```
Comment (yarikoptic) [inline .github/workflows/test.yml:51]: could have used datalad-installer to get all the "retry" etc benefits

File: src/fscacher/tests/test_cache.py (hunk 1 above)
Comment (yarikoptic) [reply src/fscacher/tests/test_cache.py:351]: hm, https://www.githubstatus.com/ and https://www.githubstatus.com/history report nothing .  Another force push may be?

Comment (yarikoptic) [reply src/fscacher/tests/test_cache.py:351]: although may be no need -- shows that this code is covered now and whatever test runs happened, are green

**Comment by yarikoptic** [comment]
Looks good to me for the original issue purpose, let's proceed, thank you @jwodder!

**Pull request merged**